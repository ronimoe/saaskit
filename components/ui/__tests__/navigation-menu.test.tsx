import { render, screen } from '@testing-library/react';
import {
  NavigationMenu,
  NavigationMenuList,
  NavigationMenuItem,
  NavigationMenuTrigger,
  NavigationMenuContent,
  NavigationMenuLink,
} from '@/components/ui/navigation-menu';

// Mock the Radix UI Navigation Menu
jest.mock('@radix-ui/react-navigation-menu', () => {
    const original = jest.requireActual('@radix-ui/react-navigation-menu');
    return {
      ...original,
      Root: ({ children }: { children: React.ReactNode }) => <nav>{children}</nav>,
      List: ({ children }: { children: React.ReactNode }) => <ul>{children}</ul>,
      Item: ({ children }: { children: React.ReactNode }) => <li>{children}</li>,
      Trigger: ({ children }: { children: React.ReactNode }) => <button>{children}</button>,
      Content: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
      Link: ({ children, ...props }: { children: React.ReactNode, [key: string]: any }) => <a {...props}>{children}</a>,
      Viewport: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
    };
  });

describe('NavigationMenu Components', () => {
  it('should render a complete navigation menu structure', () => {
    const { container } = render(
      <NavigationMenu>
        <NavigationMenuList>
          <NavigationMenuItem>
            <NavigationMenuTrigger>Trigger</NavigationMenuTrigger>
            <NavigationMenuContent>
              <NavigationMenuLink href="/">Link</NavigationMenuLink>
            </NavigationMenuContent>
          </NavigationMenuItem>
        </NavigationMenuList>
      </NavigationMenu>
    );

    expect(screen.getByText('Trigger')).toBeInTheDocument();
    expect(screen.getByText('Link')).toBeInTheDocument();
    expect(screen.getByRole('link')).toHaveAttribute('href', '/');
    expect(container).toMatchSnapshot();
  });
}); 