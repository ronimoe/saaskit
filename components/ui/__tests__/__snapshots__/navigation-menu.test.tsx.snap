// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu Components should render a complete navigation menu structure 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <button>
          Trigger
           
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-down relative top-[1px] ml-1 size-3 transition duration-300 group-data-[state=open]:rotate-180"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
        <div>
          <a
            class="data-[active=true]:focus:bg-accent data-[active=true]:hover:bg-accent data-[active=true]:bg-accent/50 data-[active=true]:text-accent-foreground hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground focus-visible:ring-ring/50 [&_svg:not([class*='text-'])]:text-muted-foreground flex flex-col gap-1 rounded-sm p-2 text-sm transition-all outline-none focus-visible:ring-[3px] focus-visible:outline-1 [&_svg:not([class*='size-'])]:size-4"
            data-slot="navigation-menu-link"
            href="/"
          >
            Link
          </a>
        </div>
      </li>
    </ul>
    <div
      class="absolute top-full left-0 isolate z-50 flex justify-center"
    >
      <div />
    </div>
  </nav>
</div>
`;
